src/test/java/com/clouway/cuse/SearchEngineContractTest.java

```diff
@@ -979,6 +979,40 @@ public void searchEmptyString() throws Exception {
     assertThat(result.size(), is(3));
   }
 
+  @Test
+  public void registerDocumentsInBulkChunks() throws Exception {
+    List<Object> employeeList = generateGiantListOfEmployeesNamedFooBar(402);
+    storeAllEmployee(employeeList);
+    store(new User(2l));
+    List result = searchEngine.search(Employee.class).returnAll().now();
+    assertThat(employeeList.size(), is(result.size()));
+  }
+
+  @Test
+  public void searchingForNonExistentEmployeeAfterBulkRegistering() throws Exception {
+    List<Object> employeeList = generateGiantListOfEmployeesNamedFooBar(201);
+    storeAllEmployee(employeeList);
+    List result = searchEngine.search(Employee.class).where("Rab").returnAll().now();
+    assertThat(result.size(), is(0));
+  }
+
+  @Test(expected = EmptyIndexListException.class)
+  public void passingEmptyListInRegisterAll() throws Exception {
+    List<Object> emptyList = Lists.newArrayList();
+    storeAllEmployee(emptyList);
+  }
+
+  //Fills a list with given size with Employees named Foo Bar.
+  private List<Object> generateGiantListOfEmployeesNamedFooBar(Integer size) {
```

Comment:
It will be more convenient to have the ability to call `registerAll` with a `List<MyClass>`.
You can try changing the registerAll argument to `List<?>` (the same as `List<? extends Object>`).